src/editor.c

```diff
@@ -697,51 +700,106 @@ static void request_reshowing_calltip(SCNotification *nt)
 }
 
 
-static void autocomplete_scope(GeanyEditor *editor)
+static gboolean autocomplete_scope(GeanyEditor *editor, const gchar *root, gsize rootlen)
 {
 	ScintillaObject *sci = editor->sci;
 	gint pos = sci_get_current_position(editor->sci);
 	gchar typed = sci_get_char_at(sci, pos - 1);
+	gchar brace_char;
 	gchar *name;
-	const GPtrArray *tags = NULL;
-	const TMTag *tag;
 	GeanyFiletype *ft = editor->document->file_type;
+	GPtrArray *tags;
+	gboolean function = FALSE;
+	gboolean member;
+	gboolean ret = FALSE;
+	const gchar *current_scope;
+	const gchar *context_sep = symbols_get_context_separator(ft->id);
```

Comment:
No no, I just meant calling `tm_tag_context_separator(ft->lang)` here instead of `symbols_get_context_separator(ft->id)` to avoid the indirection of `symbols_get_context_separator()` itself calling `tm_tag_context_separator()`.